spotify/main.py

```diff
@@ -304,8 +318,8 @@ def main():
         # Fetch aggregate data for the episode in 3x1 day changes
         # (yesterday, the day before yesterday, and the day before that)
         # Otherwise you get aggregated data of 3 days.
-        for i in range(3):
-            end = dt.datetime.now() - dt.timedelta(days=i+1) #start from yesterday
+        for i in range(days):
+            end = end_datetime - dt.timedelta(days=i) #start from yesterday
```

Comment:
should be still `dt.timedelta(days=i+1)` as there is just data starting from yesterday